### PR TITLE
vkd3d: Separate root VA use for CBV and SRV/UAV.

### PR DESCRIPTION
**Pull request by HansKristian-Work**
Signed-off-by: Hans-Kristian Arntzen <post@arntzen-software.no>

### CODE REVIEW

File: libs/vkd3d/command.c [hunk 1]
```diff
@@ -3372,26 +3372,37 @@ static void d3d12_command_list_update_root_descriptors(struct d3d12_command_list
...
         descriptor_set = d3d12_command_allocator_allocate_descriptor_set(
                 list->allocator, root_signature->vk_root_descriptor_layout, VKD3D_DESCRIPTOR_POOL_TYPE_STATIC);
     }
 
-    if (!(list->device->bindless_state.flags & VKD3D_RAW_VA_ROOT_DESCRIPTOR))
+    if (bindings->root_descriptor_dirty_mask)
     {
-        /* TODO bind null descriptors for inactive root descriptors */
-        bindings->root_descriptor_dirty_mask &= bindings->root_descriptor_active_mask;
+        /* If any push descriptor is dirty, we need to update all of them. */
```
Comment (doitsujin) [inline libs/vkd3d/command.c:3393]: Wording seems misleading here since this is about the raw VA ones, not the actual push descriptors.

**Review by doitsujin — COMMENTED** [review]
Nit with that one comment, looks fine otherwise.

File: libs/vkd3d/command.c (hunk 1 above)
Comment (HansKristian-Work) [reply libs/vkd3d/command.c:3393]: Right.

Comment (HansKristian-Work) [reply libs/vkd3d/command.c]: Fixed.

**Pull request merged**